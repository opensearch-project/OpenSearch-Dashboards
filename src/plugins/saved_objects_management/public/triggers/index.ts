/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export {
  SAVED_OBJECT_DELETE_TRIGGER,
  savedObjectDeleteTrigger,
} from './saved_object_delete_trigger';
