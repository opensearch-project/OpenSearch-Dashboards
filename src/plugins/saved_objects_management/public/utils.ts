/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export function formatWorkspaceIdParams<T extends { workspaces?: string[] | null }>(
  obj: T
): T | Omit<T, 'workspaces'> {
  const { workspaces, ...others } = obj;
  if (workspaces) {
    return obj;
  }
  return others;
}
