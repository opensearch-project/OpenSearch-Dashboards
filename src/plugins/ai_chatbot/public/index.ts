/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 */

import { AIChatbotPlugin } from './plugin';

export function plugin() {
  return new AIChatbotPlugin();
}

export { AIChatbotPlugin };
export * from './types';