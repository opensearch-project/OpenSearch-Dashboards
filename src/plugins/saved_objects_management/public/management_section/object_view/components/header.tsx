/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Any modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import {
  EuiFlexGroup,
  EuiFlexItem,
  EuiButton,
  EuiPageContentHeader,
  EuiPageContentHeaderSection,
  EuiText,
} from '@elastic/eui';
import { FormattedMessage } from '@osd/i18n/react';
import { i18n } from '@osd/i18n';
import { ApplicationStart } from '../../../../../../core/public';
import {
  NavigationPublicPluginStart,
  TopNavControlButtonData,
  TopNavControlIconData,
} from '../../../../../navigation/public';

interface HeaderProps {
  canEdit: boolean;
  canDelete: boolean;
  canViewInApp: boolean;
  type: string;
  viewUrl: string;
  onDeleteClick: () => void;
  useUpdatedUX: boolean;
  navigationUI: NavigationPublicPluginStart['ui'];
  application: ApplicationStart;
}

export const Header = ({
  canEdit,
  canDelete,
  canViewInApp,
  type,
  viewUrl,
  onDeleteClick,
  useUpdatedUX,
  navigationUI: { HeaderControl },
  application,
}: HeaderProps) => {
  return useUpdatedUX ? (
    <HeaderControl
      controls={[
        ...(canDelete
          ? [
              {
                controlType: 'icon',
                iconType: 'trash',
                ariaLabel: i18n.translate('savedObjectsManagement.view.deleteItemButtonLabel', {
                  defaultMessage: 'Delete {title}',
                  values: { title: type },
                }),
                run: onDeleteClick,
                testId: 'savedObjectEditDelete',
                display: 'base',
                color: 'danger',
              } as TopNavControlIconData,
            ]
          : []),
        ...(canViewInApp
          ? [
              {
                controlType: 'button',
                iconType: 'popout',
                label: i18n.translate('savedObjectsManagement.view.viewItemButtonLabel', {
                  defaultMessage: 'View {title}',
                  values: { title: type },
                }),
                href: viewUrl,
                testId: 'savedObjectEditViewInApp',
                fill: true,
                iconSide: 'right',
              } as TopNavControlButtonData,
            ]
          : []),
      ]}
      setMountPoint={application.setAppRightControls}
    />
  ) : (
    <EuiPageContentHeader>
      <EuiPageContentHeaderSection>
        <EuiText size="s">
          {canEdit ? (
            <h1>
              <FormattedMessage
                id="savedObjectsManagement.view.editItemTitle"
                defaultMessage="Edit {title}"
                values={{ title: type }}
              />
            </h1>
          ) : (
            <h1>
              <FormattedMessage
                id="savedObjectsManagement.view.viewItemTitle"
                defaultMessage="View {title}"
                values={{ title: type }}
              />
            </h1>
          )}
        </EuiText>
      </EuiPageContentHeaderSection>
      <EuiPageContentHeaderSection>
        <EuiFlexGroup responsive={false}>
          {canViewInApp && (
            <EuiFlexItem grow={false}>
              <EuiButton
                size="s"
                href={viewUrl}
                iconType="eye"
                data-test-subj="savedObjectEditViewInApp"
              >
                <FormattedMessage
                  id="savedObjectsManagement.view.viewItemButtonLabel"
                  defaultMessage="View {title}"
                  values={{ title: type }}
                />
              </EuiButton>
            </EuiFlexItem>
          )}
          {canDelete && (
            <EuiFlexItem grow={false}>
              <EuiButton
                color="danger"
                size="s"
                iconType="trash"
                onClick={() => onDeleteClick()}
                data-test-subj="savedObjectEditDelete"
              >
                <FormattedMessage
                  id="savedObjectsManagement.view.deleteItemButtonLabel"
                  defaultMessage="Delete {title}"
                  values={{ title: type }}
                />
              </EuiButton>
            </EuiFlexItem>
          )}
        </EuiFlexGroup>
      </EuiPageContentHeaderSection>
    </EuiPageContentHeader>
  );
};
