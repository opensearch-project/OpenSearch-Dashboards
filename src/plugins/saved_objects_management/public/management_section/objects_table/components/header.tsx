/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Any modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React, { Fragment } from 'react';
import {
  EuiSpacer,
  EuiFlexGroup,
  EuiFlexItem,
  EuiText,
  EuiTextColor,
  EuiButtonEmpty,
} from '@elastic/eui';
import { FormattedMessage } from '@osd/i18n/react';
import { ApplicationStart } from 'src/core/public';
import { i18n } from '@osd/i18n';
import {
  NavigationPublicPluginStart,
  TopNavControlButtonData,
} from '../../../../../navigation/public';

export const Header = ({
  onExportAll,
  onImport,
  onDuplicate,
  onRefresh,
  objectCount,
  showDuplicateAll = false,
  useUpdatedUX,
  navigationUI: { HeaderControl },
  applications,
  currentWorkspaceName,
  showImportButton,
}: {
  onExportAll: () => void;
  onImport: () => void;
  onDuplicate: () => void;
  onRefresh: () => void;
  objectCount: number;
  showDuplicateAll: boolean;
  useUpdatedUX: boolean;
  navigationUI: NavigationPublicPluginStart['ui'];
  applications: ApplicationStart;
  currentWorkspaceName: string;
  showImportButton: boolean;
}) => {
  const title = useUpdatedUX ? null : (
    <EuiFlexItem grow={false}>
      <EuiText size="s">
        <h1>
          <FormattedMessage
            id="savedObjectsManagement.objectsTable.header.savedObjectsTitle"
            defaultMessage="Saved Objects"
          />
        </h1>
      </EuiText>
    </EuiFlexItem>
  );
  const description = useUpdatedUX ? (
    <HeaderControl
      controls={[
        currentWorkspaceName
          ? {
              description: i18n.translate(
                'savedObjectsManagement.workspace.objectsTable.table.description',
                {
                  defaultMessage: 'Manage and share assets for {workspace}.',
                  values: {
                    workspace: currentWorkspaceName,
                  },
                }
              ),
            }
          : {
              description: i18n.translate('savedObjectsManagement.objectsTable.table.description', {
                defaultMessage: 'Manage and share your assets.',
              }),
            },
      ]}
      setMountPoint={applications.setAppDescriptionControls}
    />
  ) : (
    <EuiText size="s">
      <p>
        <EuiTextColor color="subdued">
          <FormattedMessage
            id="savedObjectsManagement.objectsTable.howToDeleteSavedObjectsDescription"
            defaultMessage="Manage and share your saved objects. To edit the underlying data of an object, go to its associated application."
          />
        </EuiTextColor>
      </p>
    </EuiText>
  );

  const rightControls = useUpdatedUX ? (
    <HeaderControl
      controls={[
        ...(showDuplicateAll
          ? [
              {
                testId: 'duplicateObjects',
                run: onDuplicate,
                controlType: 'button',
                isDisabled: objectCount === 0,
                iconType: 'copy',
                label: i18n.translate(
                  'savedObjectsManagement.objectsTable.header.duplicateAllAssetsButtonLabel',
                  { defaultMessage: 'Copy all assets to...' }
                ),
              } as TopNavControlButtonData,
            ]
          : []),
        {
          testId: 'exportAllObjects',
          run: onExportAll,
          controlType: 'button',
          iconType: 'exportAction',
          label: i18n.translate(
            'savedObjectsManagement.objectsTable.header.exportAssetsButtonLabel',
            {
              defaultMessage: 'Export all assets',
            }
          ),
        } as TopNavControlButtonData,
        ...(showImportButton
          ? [
              {
                testId: 'importObjects',
                run: onImport,
                controlType: 'button',
                iconType: 'importAction',
                label: i18n.translate(
                  'savedObjectsManagement.objectsTable.header.importButtonLabel',
                  {
                    defaultMessage: 'Import',
                  }
                ),
              } as TopNavControlButtonData,
            ]
          : []),
      ]}
      setMountPoint={applications.setAppRightControls}
    />
  ) : (
    <EuiFlexItem grow={false}>
      <EuiFlexGroup alignItems="baseline" gutterSize="m" responsive={false}>
        {showDuplicateAll && (
          <EuiFlexItem grow={false}>
            <EuiButtonEmpty
              size="s"
              data-test-subj="duplicateObjects"
              onClick={onDuplicate}
              disabled={objectCount === 0}
              iconType="copy"
            >
              <FormattedMessage
                id="savedObjectsManagement.objectsTable.header.duplicateAllButtonLabel"
                defaultMessage="Copy all objects to..."
              />
            </EuiButtonEmpty>
          </EuiFlexItem>
        )}
        <EuiFlexItem grow={false}>
          <EuiButtonEmpty
            size="s"
            iconType="exportAction"
            data-test-subj="exportAllObjects"
            onClick={onExportAll}
          >
            <FormattedMessage
              id="savedObjectsManagement.objectsTable.header.exportButtonLabel"
              defaultMessage="Export all objects"
            />
          </EuiButtonEmpty>
        </EuiFlexItem>
        {showImportButton && (
          <EuiFlexItem grow={false}>
            <EuiButtonEmpty
              size="s"
              iconType="importAction"
              data-test-subj="importObjects"
              onClick={onImport}
            >
              <FormattedMessage
                id="savedObjectsManagement.objectsTable.header.importButtonLabel"
                defaultMessage="Import"
              />
            </EuiButtonEmpty>
          </EuiFlexItem>
        )}
        <EuiFlexItem grow={false}>
          <EuiButtonEmpty size="s" iconType="refresh" onClick={onRefresh}>
            <FormattedMessage
              id="savedObjectsManagement.objectsTable.header.refreshButtonLabel"
              defaultMessage="Refresh"
            />
          </EuiButtonEmpty>
        </EuiFlexItem>
      </EuiFlexGroup>
    </EuiFlexItem>
  );

  return (
    <Fragment>
      <EuiFlexGroup justifyContent="spaceBetween" alignItems="baseline">
        {title}
        {rightControls}
      </EuiFlexGroup>
      {!useUpdatedUX ? <EuiSpacer size="m" /> : <EuiSpacer size="l" />}
      {description}
      {!useUpdatedUX && <EuiSpacer size="m" />}
    </Fragment>
  );
};
