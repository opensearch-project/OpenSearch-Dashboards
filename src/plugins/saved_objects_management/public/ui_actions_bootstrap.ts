/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { UiActionsSetup } from '../../ui_actions/public';
import { SAVED_OBJECT_DELETE_TRIGGER, savedObjectDeleteTrigger } from './triggers';

export interface SavedObjectDeleteContext {
  type: string;
  savedObjectId: string;
}

declare module '../../ui_actions/public' {
  export interface TriggerContextMapping {
    [SAVED_OBJECT_DELETE_TRIGGER]: SavedObjectDeleteContext;
  }
}

export const bootstrap = (uiActions: UiActionsSetup) => {
  uiActions.registerTrigger(savedObjectDeleteTrigger);
};
