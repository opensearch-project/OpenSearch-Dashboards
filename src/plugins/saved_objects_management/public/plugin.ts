/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Any modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import { i18n } from '@osd/i18n';
import { CoreSetup, CoreStart, Plugin } from 'src/core/public';

import { VisBuilderStart } from '../../vis_builder/public';
import { ManagementSetup } from '../../management/public';
import { UiActionsSetup, UiActionsStart } from '../../ui_actions/public';
import { DataPublicPluginStart } from '../../data/public';
import { DashboardStart } from '../../dashboard/public';
import { DiscoverStart } from '../../discover/public';
import { HomePublicPluginSetup, FeatureCatalogueCategory } from '../../home/public';
import { VisualizationsStart } from '../../visualizations/public';
import { VisAugmenterStart } from '../../vis_augmenter/public';
import {
  SavedObjectsManagementActionService,
  SavedObjectsManagementActionServiceSetup,
  SavedObjectsManagementActionServiceStart,
  SavedObjectsManagementColumnService,
  SavedObjectsManagementColumnServiceSetup,
  SavedObjectsManagementColumnServiceStart,
  SavedObjectsManagementNamespaceService,
  SavedObjectsManagementNamespaceServiceSetup,
  SavedObjectsManagementNamespaceServiceStart,
  SavedObjectsManagementServiceRegistry,
  ISavedObjectsManagementServiceRegistry,
} from './services';
import { registerServices } from './register_services';
import { bootstrap } from './ui_actions_bootstrap';

export interface SavedObjectsManagementPluginSetup {
  actions: SavedObjectsManagementActionServiceSetup;
  columns: SavedObjectsManagementColumnServiceSetup;
  namespaces: SavedObjectsManagementNamespaceServiceSetup;
  serviceRegistry: ISavedObjectsManagementServiceRegistry;
}

export interface SavedObjectsManagementPluginStart {
  actions: SavedObjectsManagementActionServiceStart;
  columns: SavedObjectsManagementColumnServiceStart;
  namespaces: SavedObjectsManagementNamespaceServiceStart;
}

export interface SetupDependencies {
  management: ManagementSetup;
  home?: HomePublicPluginSetup;
  uiActions: UiActionsSetup;
}

export interface StartDependencies {
  data: DataPublicPluginStart;
  dashboard?: DashboardStart;
  visualizations?: VisualizationsStart;
  visAugmenter?: VisAugmenterStart;
  discover?: DiscoverStart;
  visBuilder?: VisBuilderStart;
  uiActions: UiActionsStart;
}

export class SavedObjectsManagementPlugin
  implements
    Plugin<
      SavedObjectsManagementPluginSetup,
      SavedObjectsManagementPluginStart,
      SetupDependencies,
      StartDependencies
    > {
  private actionService = new SavedObjectsManagementActionService();
  private columnService = new SavedObjectsManagementColumnService();
  private namespaceService = new SavedObjectsManagementNamespaceService();
  private serviceRegistry = new SavedObjectsManagementServiceRegistry();

  public setup(
    core: CoreSetup<StartDependencies, SavedObjectsManagementPluginStart>,
    { home, management, uiActions }: SetupDependencies
  ): SavedObjectsManagementPluginSetup {
    const actionSetup = this.actionService.setup();
    const columnSetup = this.columnService.setup();
    const namespaceSetup = this.namespaceService.setup();

    if (home) {
      home.featureCatalogue.register({
        id: 'saved_objects',
        title: i18n.translate('savedObjectsManagement.objects.savedObjectsTitle', {
          defaultMessage: 'Saved Objects',
        }),
        description: i18n.translate('savedObjectsManagement.objects.savedObjectsDescription', {
          defaultMessage:
            'Import, export, and manage your saved searches, visualizations, and dashboards.',
        }),
        icon: 'savedObjectsApp',
        path: '/app/management/opensearch-dashboards/objects',
        showOnHomePage: false,
        category: FeatureCatalogueCategory.ADMIN,
      });
    }

    const opensearchDashboardsSection = management.sections.section.opensearchDashboards;
    opensearchDashboardsSection.registerApp({
      id: 'objects',
      title: i18n.translate('savedObjectsManagement.managementSectionLabel', {
        defaultMessage: 'Saved objects',
      }),
      order: 1,
      mount: async (mountParams) => {
        const { mountManagementSection } = await import('./management_section');
        return mountManagementSection({
          core,
          serviceRegistry: this.serviceRegistry,
          mountParams,
        });
      },
    });

    // sets up the context mappings and registers any triggers/actions for the plugin
    bootstrap(uiActions);

    // depends on `getStartServices`, should not be awaited
    registerServices(this.serviceRegistry, core.getStartServices);

    return {
      actions: actionSetup,
      columns: columnSetup,
      namespaces: namespaceSetup,
      serviceRegistry: this.serviceRegistry,
    };
  }

  public start(core: CoreStart, { data, uiActions }: StartDependencies) {
    const actionStart = this.actionService.start();
    const columnStart = this.columnService.start();
    const namespaceStart = this.namespaceService.start();

    return {
      actions: actionStart,
      columns: columnStart,
      namespaces: namespaceStart,
    };
  }
}
