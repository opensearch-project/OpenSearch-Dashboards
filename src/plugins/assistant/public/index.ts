/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import './index.scss';

import { PluginInitializerContext } from '../../../core/public';
import { AssistantConfig } from '../common/config';
import { AssistantPlugin } from './plugin';

// This exports static code and TypeScript types,
// as well as, OpenSearch Dashboards Platform `plugin()` initializer.
export function plugin(initializerContext: PluginInitializerContext<AssistantConfig>) {
  return new AssistantPlugin(initializerContext);
}
export { AssistantPluginSetup, AssistantPluginStart } from './types';
