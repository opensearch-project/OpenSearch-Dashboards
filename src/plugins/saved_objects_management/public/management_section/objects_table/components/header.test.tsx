/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Any modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './header';
import { applicationServiceMock } from '../../../../../../core/public/mocks';

const defaultProps = {
  onExportAll: () => {},
  onImport: () => {},
  onRefresh: () => {},
  onDuplicate: () => {},
  objectCount: 4,
  filteredCount: 2,
  useUpdatedUX: false,
  navigationUI: { HeaderControl: () => null, TopNavMenu: () => null },
  applications: applicationServiceMock.createStartContract(),
};

describe('Header', () => {
  it('should render normally', () => {
    const props = {
      ...defaultProps,
      showDuplicateAll: false,
    };

    const component = shallow(<Header {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('should render normally when showDuplicateAll is undefined', () => {
    const props = {
      ...defaultProps,
      showDuplicateAll: undefined,
    };

    const component = shallow(<Header {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('should render normally when useUpdatedUX is true', () => {
    const props = {
      ...defaultProps,
      showDuplicateAll: true,
      useUpdatedUX: true,
    };

    const component = shallow(<Header {...props} />);

    expect(component).toMatchSnapshot();
  });
});

describe('Header - workspace enabled', () => {
  it('should render `Duplicate All` button when workspace enabled', () => {
    const props = {
      ...defaultProps,
      showDuplicateAll: true,
    };

    const component = shallow(<Header {...props} />);

    expect(component.find('EuiButtonEmpty[data-test-subj="duplicateObjects"]').exists()).toBe(true);
  });
});
