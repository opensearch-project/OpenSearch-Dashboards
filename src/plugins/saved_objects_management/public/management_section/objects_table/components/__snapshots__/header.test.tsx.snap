// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header - workspace enabled should render \`Import\` button inside a workspace 1`] = `
<Fragment>
  <EuiFlexGroup
    alignItems="baseline"
    justifyContent="spaceBetween"
  >
    <HeaderControl
      controls={
        Array [
          Object {
            "controlType": "button",
            "iconType": "exportAction",
            "label": "Export all assets",
            "run": [Function],
            "testId": "exportAllObjects",
          },
          Object {
            "controlType": "button",
            "iconType": "importAction",
            "label": "Import",
            "run": [Function],
            "testId": "importObjects",
          },
        ]
      }
      setMountPoint={[MockFunction]}
    />
  </EuiFlexGroup>
  <EuiSpacer
    size="l"
  />
  <HeaderControl
    controls={
      Array [
        Object {
          "description": "Manage and share your assets.",
        },
      ]
    }
    setMountPoint={[MockFunction]}
  />
</Fragment>
`;

exports[`Header should render normally 1`] = `
<Fragment>
  <EuiFlexGroup
    alignItems="baseline"
    justifyContent="spaceBetween"
  >
    <EuiFlexItem
      grow={false}
    >
      <EuiText
        size="s"
      >
        <h1>
          <FormattedMessage
            defaultMessage="Saved Objects"
            id="savedObjectsManagement.objectsTable.header.savedObjectsTitle"
            values={Object {}}
          />
        </h1>
      </EuiText>
    </EuiFlexItem>
    <EuiFlexItem
      grow={false}
    >
      <EuiFlexGroup
        alignItems="baseline"
        gutterSize="m"
        responsive={false}
      >
        <EuiFlexItem
          grow={false}
        >
          <EuiButtonEmpty
            data-test-subj="exportAllObjects"
            iconType="exportAction"
            onClick={[Function]}
            size="s"
          >
            <FormattedMessage
              defaultMessage="Export all objects"
              id="savedObjectsManagement.objectsTable.header.exportButtonLabel"
              values={Object {}}
            />
          </EuiButtonEmpty>
        </EuiFlexItem>
        <EuiFlexItem
          grow={false}
        >
          <EuiButtonEmpty
            data-test-subj="importObjects"
            iconType="importAction"
            onClick={[Function]}
            size="s"
          >
            <FormattedMessage
              defaultMessage="Import"
              id="savedObjectsManagement.objectsTable.header.importButtonLabel"
              values={Object {}}
            />
          </EuiButtonEmpty>
        </EuiFlexItem>
        <EuiFlexItem
          grow={false}
        >
          <EuiButtonEmpty
            iconType="refresh"
            onClick={[Function]}
            size="s"
          >
            <FormattedMessage
              defaultMessage="Refresh"
              id="savedObjectsManagement.objectsTable.header.refreshButtonLabel"
              values={Object {}}
            />
          </EuiButtonEmpty>
        </EuiFlexItem>
      </EuiFlexGroup>
    </EuiFlexItem>
  </EuiFlexGroup>
  <EuiSpacer
    size="m"
  />
  <EuiText
    size="s"
  >
    <p>
      <EuiTextColor
        color="subdued"
      >
        <FormattedMessage
          defaultMessage="Manage and share your saved objects. To edit the underlying data of an object, go to its associated application."
          id="savedObjectsManagement.objectsTable.howToDeleteSavedObjectsDescription"
          values={Object {}}
        />
      </EuiTextColor>
    </p>
  </EuiText>
  <EuiSpacer
    size="m"
  />
</Fragment>
`;

exports[`Header should render normally when showDuplicateAll is undefined 1`] = `
<Fragment>
  <EuiFlexGroup
    alignItems="baseline"
    justifyContent="spaceBetween"
  >
    <EuiFlexItem
      grow={false}
    >
      <EuiText
        size="s"
      >
        <h1>
          <FormattedMessage
            defaultMessage="Saved Objects"
            id="savedObjectsManagement.objectsTable.header.savedObjectsTitle"
            values={Object {}}
          />
        </h1>
      </EuiText>
    </EuiFlexItem>
    <EuiFlexItem
      grow={false}
    >
      <EuiFlexGroup
        alignItems="baseline"
        gutterSize="m"
        responsive={false}
      >
        <EuiFlexItem
          grow={false}
        >
          <EuiButtonEmpty
            data-test-subj="exportAllObjects"
            iconType="exportAction"
            onClick={[Function]}
            size="s"
          >
            <FormattedMessage
              defaultMessage="Export all objects"
              id="savedObjectsManagement.objectsTable.header.exportButtonLabel"
              values={Object {}}
            />
          </EuiButtonEmpty>
        </EuiFlexItem>
        <EuiFlexItem
          grow={false}
        >
          <EuiButtonEmpty
            data-test-subj="importObjects"
            iconType="importAction"
            onClick={[Function]}
            size="s"
          >
            <FormattedMessage
              defaultMessage="Import"
              id="savedObjectsManagement.objectsTable.header.importButtonLabel"
              values={Object {}}
            />
          </EuiButtonEmpty>
        </EuiFlexItem>
        <EuiFlexItem
          grow={false}
        >
          <EuiButtonEmpty
            iconType="refresh"
            onClick={[Function]}
            size="s"
          >
            <FormattedMessage
              defaultMessage="Refresh"
              id="savedObjectsManagement.objectsTable.header.refreshButtonLabel"
              values={Object {}}
            />
          </EuiButtonEmpty>
        </EuiFlexItem>
      </EuiFlexGroup>
    </EuiFlexItem>
  </EuiFlexGroup>
  <EuiSpacer
    size="m"
  />
  <EuiText
    size="s"
  >
    <p>
      <EuiTextColor
        color="subdued"
      >
        <FormattedMessage
          defaultMessage="Manage and share your saved objects. To edit the underlying data of an object, go to its associated application."
          id="savedObjectsManagement.objectsTable.howToDeleteSavedObjectsDescription"
          values={Object {}}
        />
      </EuiTextColor>
    </p>
  </EuiText>
  <EuiSpacer
    size="m"
  />
</Fragment>
`;

exports[`Header should render normally when useUpdatedUX is true 1`] = `
<Fragment>
  <EuiFlexGroup
    alignItems="baseline"
    justifyContent="spaceBetween"
  >
    <HeaderControl
      controls={
        Array [
          Object {
            "controlType": "button",
            "iconType": "copy",
            "isDisabled": false,
            "label": "Copy all assets to...",
            "run": [Function],
            "testId": "duplicateObjects",
          },
          Object {
            "controlType": "button",
            "iconType": "exportAction",
            "label": "Export all assets",
            "run": [Function],
            "testId": "exportAllObjects",
          },
          Object {
            "controlType": "button",
            "iconType": "importAction",
            "label": "Import",
            "run": [Function],
            "testId": "importObjects",
          },
        ]
      }
      setMountPoint={[MockFunction]}
    />
  </EuiFlexGroup>
  <EuiSpacer
    size="l"
  />
  <HeaderControl
    controls={
      Array [
        Object {
          "description": "Manage and share your assets.",
        },
      ]
    }
    setMountPoint={[MockFunction]}
  />
</Fragment>
`;
